src/browser.ts

```diff
@@ -85,18 +99,28 @@ export class GrpcClient {
 
       const url = `https://${servicePath}:${servicePort}/$rpc/${serviceName}.${rpcNamespace}/${rpcName}`;
 
-      const fetchResult = await fetch(url, {
+      fetch(url, {
         headers,
         method: 'post',
         body: requestData,
-      });
-
-      if (!fetchResult.ok) {
-        callback(new Error(fetchResult.statusText));
-        return;
-      }
-      const responseArrayBuffer = await fetchResult.arrayBuffer();
-      callback(null, new Uint8Array(responseArrayBuffer));
+        signal: cancelSignal,
+      })
+        .then(response => {
+          return response.arrayBuffer();
+        })
+        .then(buffer => {
+          callback(null, new Uint8Array(buffer));
+        })
+        .catch(err => {
+          if (
+            cancelHandler.cancelRequested === false &&
```

Comment:
just do `!cancelHander.cancelRequested`

Reply:
if `cancelRequested` AND it's an `AbortError` than we are good. So the opposite would be `!cancelRequested` OR `not an AbortError` :)

Reply:
Do you mean `!cancelHandler.cancelRequested || err.name !== 'AbortError'`?

Reply:
Yes, I think that's right. An exception means error and needs to be thrown if at least one of the following is true:
- cancellation was not requested
- error is not `AbortError`

Reply:
Makes sense, done.